pkg/controller/nodes/executor.go

```diff
@@ -38,6 +38,9 @@ type nodeMetrics struct {
 	ResolutionFailure  labeled.Counter
 	InputsWriteFailure labeled.Counter
 	TimedOutFailure    labeled.Counter
+	SystemFailure      labeled.Counter
```

Comment:
We already have FailureLatency metric, can we just tag that with System or not?

Reply:
The FailureDuration metric which we have today measures end-2-end latency from start of the node to when its actually marked failed. To achieve that, its emitted when we change the phase from Failing to Failed ... basically in the next round and at that time, we dont have error information available in nodestatus. I do feel that we should rework some of that logic.  

I can set up some time next week to talk about it. Please let me know @EngHabu

Reply:
Ic... Can we then rename this metric to Failure and have a label of failure type and set it to either System or User before Incrementing?

Reply:
Setting labels during increment is tricky. 

In the current model (promutils in stdlib), Labeled metrics get label values from context and the keys for those labels are defined-once through SetMetricKeys. That is being set in main.go with values: ProjectKey, DomainKey, WorkflowIDKey, TaskIDKey. So when we incr(), we cannot say system/user/etc unless we add errorKind key to the predefined labels in main.go which I dont want to do :(. @EngHabu  

I dont like this structure but I dont want to work on those issues in this PR.

Reply:
I understand. Can you get rid of the systemFailures counter then, file an issue to support that in stdlib and assign it to me?

Reply:
Done. Thanks @EngHabu